internal/vault/sys/password_policy.go

```diff
@@ -0,0 +1,269 @@
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the password policy.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"policy": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The password policy document",
+			},
+		},
+		MarkdownDescription: "Provides a resource to manage Password Policies.",
+	}
+
+	base.MustAddLegacyBaseSchema(&resp.Schema)
+}
+
+// Create is called during the terraform apply command.
+//
+// https://developer.hashicorp.com/terraform/plugin/framework/resources/create
+func (r *PasswordPolicyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data PasswordPolicyModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := client.GetClient(ctx, r.Meta(), data.Namespace.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Error Configuring Resource Client", err.Error())
+		return
+	}
+
+	vaultRequest := map[string]interface{}{
+		"policy": data.Policy.ValueString(),
+	}
+	path := r.path(data.Name.ValueString())
+	// vault returns a nil response on success
+	_, err = client.Logical().Write(path, vaultRequest)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to Create Resource",
+			"An unexpected error occurred while attempting to create the resource. "+
+				"Please retry the operation or report this issue to the provider developers.\n\n"+
+				"HTTP Error: "+err.Error(),
+		)
+
+		return
+	}
+
+	// write the ID to state which is required for backwards compatibility
+	data.ID = types.StringValue(data.Name.ValueString())
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+// Read is called during the terraform apply, terraform plan, and terraform
+// refresh commands.
+//
+// https://developer.hashicorp.com/terraform/plugin/framework/resources/read
+func (r *PasswordPolicyResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data PasswordPolicyModel
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := client.GetClient(ctx, r.Meta(), data.Namespace.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Error Configuring Resource Client", err.Error())
+		return
+	}
+
+	// read the name from the id field to support the import command
+	name := data.ID.ValueString()
+	path := r.path(name)
+	policyResp, err := client.Logical().Read(path)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to Read Resource from Vault",
```

Comment:
nit: Maybe a future improvement if we don't have time, but it would be great to make these error strings reusable by all the resource packages by defining them somewhere like `internal/framework/errutil`.

and then we could have:

```go
resp.Diagnostics.AddError(errutil.VaultReadErr(err))

// ...

resp.Diagnostics.AddError(errutil.VaultReadResponseNil(err))

// etc...
```

Reply:
Good call, added a `errutil` package to reuse errors